futures-test/src/task/wake_counter.rs

```diff
@@ -68,3 +68,52 @@ impl Wake for Inner {
         arc_self.count.fetch_add(1, Ordering::SeqCst);
     }
 }
+
+/// Number of times the waker was awoken.
+///
+/// See [`new_count_waker`] for usage.
+#[derive(Debug)]
+pub struct AwokenCount {
+    inner: Arc<WakerInner>,
+}
+
+impl PartialEq<usize> for AwokenCount {
```

Comment:
I think `Deref<Target = usize>` would also be handy.

Reply:
How can I do that? I don't think `AtomicUsize` has any methods that allow you to get a reference to the inner usize. I tried the following

```rust
impl Deref for AwokenCount {
    type Target = usize;

    fn deref(&self) -> &usize {
        &self.inner.count.load(Ordering::SeqCst)
    }
}
```

But that gives the `cannot return reference to temporary value` error, which makes sense.

I could add back the `get(&self) -> usize` method